lib/rpmlock.cc

```diff
@@ -124,6 +124,8 @@ rpmlock rpmlockNew(const char *lock_path, const char *descr)
     if (!lock) {
 	rpmlog(RPMLOG_ERR, _("can't create %s lock on %s (%s)\n"), 
 		descr, lock_path, strerror(errno));
+	if (errno == ENOENT && geteuid() != 0)
+	    rpmlog(RPMLOG_ERR, _("\trun rpm as root to create the lock file\n"));
```

Comment:
No, this is misleading and wrong. Root is not required to create locks, it needs a write permission to the directory, which might be owned by whoever. For system rpmdb it's root normally of course.

Reply:
So yeah, just drop the extra error. This is way too many assumptions.